app/operations/event_procedures/total_amount_cents.rb

```diff
@@ -2,32 +2,61 @@
 
 module EventProcedures
   class TotalAmountCents < Actor
-    input :user_id, type: Integer
-    input :month, type: String, allow_nil: true
-    input :year, type: String, allow_nil: true
+    input :event_procedures
 
     output :total, type: String
     output :payd, type: String
     output :unpaid, type: String
 
     def call
-      self.total = calculate_total
-      self.payd = calculate_paid
-      self.unpaid = calculate_unpaid
+      procedures = Procedure.where(id: event_procedures.pluck(:procedure_id))
+
+      self.total = calculate_total(procedures)
+      self.payd = calculate_payd(procedures)
+      self.unpaid = calculate_unpayd(procedures)
     end
 
-    private
+    def convert_money(amount_cents)
+      Money.new(amount_cents, "BRL").format
+    end
 
-    def calculate_total
-      Money.new(SumAmountQuery.call(user_id:, month:, year:), "BRL").format
+    def calculate_total(user_procedures)
+      procedures_by_id = {}
+      user_procedures.each do |procedure|
+        procedures_by_id[procedure.id] = procedure
+      end
+      total = 0
+      event_procedures.each do |event_procedure|
+        procedure = procedures_by_id[event_procedure.procedure_id]
+        total += procedure.amount_cents if procedure
+      end
+      convert_money(total)
     end
 
-    def calculate_paid
-      Money.new(SumAmountQuery.call(user_id:, month:, year:, payd: true), "BRL").format
+    def calculate_payd(user_procedures)
```

Comment:
The code is well structured, but there are some opportunities for improvement in readability and performance.
1. Avoid code repetition: we can use a single method to perform these calculations.
2. Replace: each + +=, with a .sum { |item| item.value }
3. Use index_by(&:id): Builds a hash { id => procedure } efficiently and avoids unnecessary repetitions.

Reply:
I thought of something like that. What do you think?

    def call
      procedures_by_id = Procedure.where(id: event_procedures.pluck(:procedure_id)).index_by(&:id)

      self.total = calculate_amount(event_procedures, procedures_by_id)
      self.payd = calculate_amount(event_procedures.select(&:payd), procedures_by_id)
      self.unpaid = calculate_amount(event_procedures.reject(&:payd), procedures_by_id)
    end

    private

    def convert_money(amount_cents)
      Money.new(amount_cents, "BRL").format
    end

    def calculate_amount(filtered_event_procedures, procedures_by_id)
      total_cents = filtered_event_procedures.sum { |event_procedure| procedures_by_id[event_procedure.procedure_id]&.amount_cents.to_i }
      convert_money(total_cents)
    end
  end